src/test/java/stepDefinition/CategoryPageStepDef.java

```diff
@@ -1,27 +1,26 @@
 package stepDefinition;
 
 import cucumber.api.java.en.When;
+import helpers.ScenarioContext;
 import org.openqa.selenium.By;
 import org.openqa.selenium.WebElement;
-import pages.CategoryPage;
 import pages.SubCategoryPage;
 
-import static helpers.Reflection.getElementByName;
-import static helpers.Reflection.getPage;
-import static utils.TestUtils.getPageObjectName;
+import static helpers.ElementSearchUtils.getElementByName;
 
 public class CategoryPageStepDef extends AbstractStepDef {
 
     protected static String expectedSubCategory;
 
-    @When("User clicks on '(.*)' subcategory")
+    @When("^user clicks on '(.*)' subcategory$")
     public void clickOnSubCategory(String subCategoryName) {
-        WebElement subCategorySelector = webDriver.findElement(By.cssSelector(".title a[href='/" + subCategoryName + "']"));
-        waitUntilElementIsClickable(subCategorySelector);
+        WebElement subCategorySelector = getElementByName("subCategoryForm")
+                .findElement(By.cssSelector(".title a[href='/" + subCategoryName + "']"));
```

Comment:
what is the purpose of this line?

Reply:
Done.
Thank You!

Reply:
anyway, line 17 and 18 is the duplicated on line 21 and 22